api/ryml.i

```diff
@@ -200,6 +207,10 @@ def parse(buf, **kwargs):
     return _call_parse(parse_csubstr, buf, **kwargs)
 
 
+def emit(tree):
+    return emit_std_string(tree)
```

Comment:
There's also the question of idiom. Most of the idea in the current version is using memory views to avoid slow allocations, but this proposes allocating memory every call. Not entirely convinced here.

Reply:
What do you propose instead?  The user could pass a bytearray into the emit function, but then SWIG code would be needed to wrap the bytearray so that the C++ rapidyaml code could control the underlying allocation.  The maintenance burden in that code vs allocating one additional time per `emit` invocation seems like a reasonable trade-off.

Reply:
I agree that the trade-off is generally reasonable; the point is that the user should be able to make the choice.

And considering that, I actually think your proposal of allocating is fine (just not `std::string`). We only have to provide another function, eg `emit_in_place()` which receives a `bytearray` that can be mutated. (This is not a request for correction, it's just me mapping how to enable power users with the capacity to avoid unnecessary allocations.)

Reply:
I've added the `emit_in_place` stuff.